java-vtl-script/src/test/java/no/ssb/vtl/script/operations/FilterOperationTest.java

```diff
@@ -20,23 +20,21 @@
  * =========================LICENSE_END==================================
  */
 
-import no.ssb.vtl.model.Component;
-import no.ssb.vtl.model.Dataset;
-import no.ssb.vtl.model.StaticDataset;
-import no.ssb.vtl.model.VTLBoolean;
-import no.ssb.vtl.model.VTLExpression;
-import no.ssb.vtl.model.VTLObject;
+import no.ssb.vtl.model.*;
 import no.ssb.vtl.script.operations.join.ComponentBindings;
+import no.ssb.vtl.script.support.DatasetCloseWatcher;
 import org.junit.Before;
 import org.junit.Test;
 
 import javax.script.Bindings;
 
+import java.util.stream.Stream;
+
 import static org.assertj.core.api.Assertions.assertThat;
 
 public class FilterOperationTest {
 
-    private Dataset dataset;
+    private DatasetCloseWatcher dataset;
```

Comment:
Nice idea with the watcher, but it leaves me wondering if this variable here is a dataset or a watcher. Maybe have the new methods under Dataset interface?

Reply:
Yeah I see your point, this only for testing though. If we add this check inside the script package we could think of a better place.